File: java/LocalBroadcastJava/app/src/main/java/com/singularitycoder/localbroadcastjava/MyJobIntentService.java
package com.singularitycoder.localbroadcastjava;

import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.core.app.JobIntentService;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import java.util.Date;

import static com.singularitycoder.localbroadcastjava.AppConstants.INTENT_ACTION_DATE;
import static com.singularitycoder.localbroadcastjava.AppConstants.INTENT_DATA_DATE;

public final class MyJobIntentService extends JobIntentService {

    static void enqueueMyWork(@NonNull final Context context, @NonNull final Integer jobId, @NonNull final Intent intent) {
        enqueueWork(context, MyJobIntentService.class, jobId, intent);
    }

    @Override
    protected void onHandleWork(@NonNull Intent intent) {
        // Send local broadcast
        final Intent myIntent = new Intent(INTENT_ACTION_DATE).putExtra(INTENT_DATA_DATE, new Date().toString());
        LocalBroadcastManager.getInstance(this).sendBroadcast(myIntent);
    }
}

File: kotlin/LocalBroadcastKotlin/app/src/main/java/com/singularitycoder/localbroadcastkotlin/MyJobIntentService.kt
package com.singularitycoder.localbroadcastkotlin

import android.content.Context
import android.content.Intent
import androidx.core.app.JobIntentService
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.singularitycoder.localbroadcastkotlin.AppConstants.INTENT_ACTION_DATE
import com.singularitycoder.localbroadcastkotlin.AppConstants.INTENT_DATA_DATE
import java.util.*

class MyJobIntentService : JobIntentService() {

    companion object {
        fun enqueueMyWork(context: Context, jobId: Int, intent: Intent) {
            enqueueWork(context, MyJobIntentService::class.java, jobId, intent)
        }
    }

    override fun onHandleWork(intent: Intent) {
        // Send local broadcast
        val myIntent: Intent = Intent(INTENT_ACTION_DATE).putExtra(INTENT_DATA_DATE, Date().toString())
        LocalBroadcastManager.getInstance(this).sendBroadcast(myIntent)
    }
}